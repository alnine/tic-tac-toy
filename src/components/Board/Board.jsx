import React from 'react';
import PropTypes from 'prop-types';
import { Cell } from '../Cell';
import StyledBoard from './Board.style';

const Board = ({ plan, onCellClick }) => {
  return (
    <StyledBoard>
      {plan &&
        plan.map((item, index) => (
          <Cell key={item.id} value={item.value} onClick={() => onCellClick(index)} />
        ))}
    </StyledBoard>
  );
};

export default Board;

Board.propTypes = {
  plan: PropTypes.arrayOf(PropTypes.object),
  onCellClick: PropTypes.func,
};

Board.defaultProps = {
  plan: null,
  onCellClick: null,
};
