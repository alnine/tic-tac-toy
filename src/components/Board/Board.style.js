import styled from 'styled-components';

const StyledBoard = styled.div`
  display: flex;
  flex-wrap: wrap;
  width: 6em;
  font-size: 1.5em;
`;

export default StyledBoard;
