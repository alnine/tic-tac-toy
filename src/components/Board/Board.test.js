import React from 'react';
import renderer from 'react-test-renderer';
import Board from './Board';

it('render correctly board component', () => {
  const BoardComponent = renderer.create(<Board />).toJSON();
  expect(BoardComponent).toMatchSnapshot();
});
