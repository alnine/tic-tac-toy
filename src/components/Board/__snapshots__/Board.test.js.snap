// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly board component 1`] = `
<div
  className="sc-AxiKw eYehTe"
/>
`;
